### PR TITLE
Groupby tuples

### PR DESCRIPTION
**Pull request by toobaz**
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

Sorry for having missed #17996, where this could have been discussed, but this said I don't think we should treat specially the case of a ``MultiIndex`` (see the test I added with tuples as labels): we should just interpret what is not a list as a label, as the docs already state (although with some incoherence, which I tried to address).

While I was at it, I fixed a typo which made a test inserted in #17996 trivial.

### CODE REVIEW

File: pandas/core/groupby.py [hunk 1]
```diff
@@ -2765,9 +2765,7 @@ def _get_grouper(obj, key=None, axis=0, level=None, sort=True,
     elif isinstance(key, BaseGrouper):
         return key, [], obj
 
-    # when MultiIndex, allow tuple to be a key
-    if not isinstance(key, (tuple, list)) or \
-            (isinstance(key, tuple) and is_axis_multiindex):
+    if not isinstance(key, list):
```
Comment (jreback) [inline pandas/core/groupby.py]: can you add a nice comment here, explaining

File: pandas/core/generic.py [hunk 2]
```diff
@@ -5092,14 +5092,14 @@ def groupby(self, by=None, axis=0, level=None, as_index=True, sort=True,
...
+        by : mapping, function, label, or list of labels
             Used to determine the groups for the groupby.
             If ``by`` is a function, it's called on each value of the object's
             index. If a dict or Series is passed, the Series or dict VALUES
             will be used to determine the groups (the Series' values are first
             aligned; see ``.align()`` method). If an ndarray is passed, the
-            values are used as-is determine the groups. A str or list of strs
-            may be passed to group by the columns in ``self``
+            values are used as-is determine the groups. A label or list of
+            labels may be passed to group by the columns in ``self``.
```
Comment (jreback) [inline pandas/core/generic.py]: maybe callout what a tuple does.

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
linting issue

**Comment by toobaz** [comment]
> linting issue

Fixed! ping @jreback

**Comment by jreback** [comment]
thanks!

**Pull request closed without merging**